src/main/scala/org/kframework/minikore/Constructors.scala

```diff
@@ -0,0 +1,97 @@
+package org.kframework.minikore
+
+import org.kframework.minikore.MiniKore._
+
+
+// Constructor Factory.
```

Comment:
@grosu At the moment I'm using only traits, and no abstract classes. I like the idea of having the "main" classes as abstract classes instead of traits. However, most of the java backend Kil classes that'll extend the main kore abstract classes already extend other (unrelated to kore) abstract classes. A Java class can only extend one abstract class, but can implement multiple traits, so it'll be easier to make the kil classes implement (mini)kore if we use traits everywhere. As far as interoperability with traits goes, @cos suggested the issue was fixed in java8. 

As far as constructors are concerned, I'm ok with them being in the same place as the destructors.